Tests/Provider/VimeoProviderTest.php

```diff
@@ -147,6 +147,30 @@ public function testTransformWithUrl()
         $this->assertEquals('012341231', $media->getProviderReference(), '::getProviderReference() is set');
     }
 
+    public function testTransformWithPlayerUrl()
+    {
+        $response = new Response();
+        $response->setContent(file_get_contents(__DIR__.'/../fixtures/valid_vimeo.txt'));
+
+        $browser = $this->getMockBuilder('Buzz\Browser')->getMock();
+        $browser->expects($this->once())->method('get')->will($this->returnValue($response));
+
+        $provider = $this->getProvider($browser);
+
+        $provider->addFormat('big', array('width' => 200, 'height' => 100, 'constraint' => true));
+
+        $media = new Media();
+        $media->setBinaryContent('http://player.vimeo.com/video/012341231');
```

Comment:
You should make a generic function and use `dataProvider` to test multiple cases like `player.vimeo.com/video/123456` and `vimeo.com/123456`.